### PR TITLE
Implement asynchronous support in ODataJsonLightServiceDocumentDeserializer

### PR DESCRIPTION
**Pull request by gathogojr**
### Issues

This pull request partially fulfills #2019.

### Description

Implement asynchronous support in **`ODataJsonLightServiceDocumentDeserializer`**.

### Checklist (Uncheck if it is not completed)

- [x] *Test cases added*
- [ ] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

File: src/Microsoft.OData.Core/JsonLight/ODataJsonLightServiceDocumentDeserializer.cs [hunk 1]
```diff
@@ -312,48 +304,333 @@ private ODataServiceDocumentElement ReadServiceDocumentElement(PropertyAndAnnota
...
+
+        /// <summary>
+        /// Validates that the specified required property in the service document element has been read and value is not null.
+        /// </summary>
+        /// <param name="property">The property.</param>
+        /// <param name="propertyValues">Property values for the specified property.</param>
+        private static void ValidateServiceDocumentElementHasRequiredProperty(string property, string[] propertyValues)
+        {
+            Debug.Assert(propertyValues != null && propertyValues.Length > 0, $"{nameof(propertyValues)} != null && {nameof(propertyValues)}.Length > 0");
+
```
Comment (marabooy) [inline src/Microsoft.OData.Core/JsonLight/ODataJsonLightServiceDocumentDeserializer.cs:629]: Suggestion: 
``` 
            Debug.Assert(propertyValues != null && propertyValues.Length > 0, $"{property} != null && {property}.Length > 0");
````

File: src/Microsoft.OData.Core/JsonLight/ODataJsonLightServiceDocumentDeserializer.cs [hunk 2]
```diff
@@ -312,48 +304,333 @@ private ODataServiceDocumentElement ReadServiceDocumentElement(PropertyAndAnnota
...
         }
+
+        /// <summary>
+        /// Validates that the specified property in the service document element is not repeated.
+        /// </summary>
+        /// <param name="property">The property.</param>
+        /// <param name="propertyValues">Property values for the specified property.</param>
+        private static void ValidateServiceDocumentElementHasNoRepeatedProperty(string property, string[] propertyValues)
+        {
+            Debug.Assert(propertyValues != null && propertyValues.Length > 0, $"{nameof(propertyValues)} != null && {nameof(propertyValues)}.Length > 0");
```
Comment (marabooy) [inline src/Microsoft.OData.Core/JsonLight/ODataJsonLightServiceDocumentDeserializer.cs:613]: Refer to the same from `ValidateServiceDocumentElementHasRequiredProperty`

File: src/Microsoft.OData.Core/JsonLight/ODataJsonLightServiceDocumentDeserializer.cs (hunk 1 above)
Comment (gathogojr) [reply src/Microsoft.OData.Core/JsonLight/ODataJsonLightServiceDocumentDeserializer.cs:629]: I have a string array `string[] propertyValues`
This:
```csharp
$"{propertyValues} != null && {propertyValues}.Length > 0"
```
gives me `" != null && .Length > 0"`
Which is not what I'm looking for.

On the other hand,
```csharp
$"{nameof(propertyValues)} != null && {nameof(propertyValues)}.Length > 0"
```
gives me `"propertyValues != null && propertyValues.Length > 0"`

File: src/Microsoft.OData.Core/JsonLight/ODataJsonLightServiceDocumentDeserializer.cs (hunk 2 above)
Comment (gathogojr) [reply src/Microsoft.OData.Core/JsonLight/ODataJsonLightServiceDocumentDeserializer.cs:613]: See my comment here https://github.com/OData/odata.net/pull/2322#discussion_r812578515

**Review by KenitoInc — APPROVED** [review]
LGTM

**Review by marabooy — APPROVED** [review]

**Pull request merged**